anchor/anchorRecord.go

```diff
@@ -96,24 +110,49 @@ func (ar *AnchorRecord) MarshalAndSignV2(priv interfaces.Signer) ([]byte, []byte
 	return data, sig.Bytes(), nil
 }
 
-func (ar *AnchorRecord) Unmarshal(data []byte) error {
+// Non-exported, refactored function that splits the AnchorRecord and its signature
```

Comment:
Is specifying "Non-exported" in the comment necessary? I don't think it conveys any additional information to the reader.

Reply:
Clarified comments to what I had intended, that its non-exported for now because it was created during refactor, but there's no reason it couldn't be exported in the future.

Reply:
agreed with paul, golang is self-documenting in that way and saying that it's not exported in a comment is not needed. documenting that a function was changed from exported to unexported should be in the commit log or change log or somewhere else (if it's an important change that affects other packages)

i would personally comment it as something like `// helper function that splits the AnchorRecord and its signature`

Reply:
My comment was really supposed to be why its non-exported. And the reason is just that I chose not to export it. Perhaps I'm thinking in too much of C++ way, but private methods in C++ are kept private as a way to control the exposed interface. So you often think carefully about what methods you expose to the outside world. My intended purpose for the comment is just that there is no good reason to have it unexported other than it was refactored an therefore is only seen internally right now and so its unexported. There's no reason it couldn't be exported in the future in the future if the functionality is useful to outside parties (not sure it is or not). At any rate. I can happily change the comment to drop all mention of non-exported, since I don't think its a big deal either way.

Reply:
I'm in favor of dropping the comment to something like @WhoSoup suggested. Documenting the obvious makes a change later now also need to touch the comment. Since the function is upper case, the comment doesn't actually stop someone from using it externally.

So if it is possible to make private, lowercasing the function method is fine. If you need it exported (for a unit test or w/e), it's fine as well.

Reply:
I removed the export comment